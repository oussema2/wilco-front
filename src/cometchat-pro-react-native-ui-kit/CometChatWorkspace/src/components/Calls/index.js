export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
