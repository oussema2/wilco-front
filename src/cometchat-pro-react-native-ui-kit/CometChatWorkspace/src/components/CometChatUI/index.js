/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
