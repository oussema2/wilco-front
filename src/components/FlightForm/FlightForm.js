/* eslint-disable react/jsx-no-bind */
import React from 'react';
import { observer } from 'mobx-react-lite';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { TextInput } from '../TextInput';
import { TouchableInput } from '../TouchableInput';
import { HorizontalPadding } from '../HorizontalPadding';
import FlightFormPresenter from '../../presenters/FlightFormPresenter';
import {
	FROM, TO, DEPARTURE_TIME, ARRIVAL_TIME
} from '../../constants/formFields/postForm';
import { styles } from './FlightForm.styles';
import { Title } from '../Title';

const FlightForm = ( { testID, presenter } ) => (
	<HorizontalPadding testID={testID} style={styles.container}>
		<Title
			testID="post-manually-form-title"
			text="Flight data - Add flight manually"
			textTransform="none"
			required
		/>
		<View style={styles.separatorView} />
		<TextInput
			testID="from-input"
			textInputStyle={styles.withBottomSpace}
			required
			autoCapitalize="characters"
			error={presenter.form.$( FROM ).error}
			inputProps={presenter.form.$( FROM ).bind()}
		/>
		<TextInput
			testID="to-input"
			textInputStyle={styles.withBottomSpace}
			required
			autoCapitalize="characters"
			error={presenter.form.$( TO ).error}
			inputProps={presenter.form.$( TO ).bind()}
		/>
		<TouchableInput
			testID="departure-time-input"
			containerStyle={styles.withBottomSpace}
			required
			label={presenter.form.$( DEPARTURE_TIME ).label}
			value={presenter.departureTime}
			onPress={() => presenter.onDepartureTimePressed()}
			placeholder={presenter.form.$( DEPARTURE_TIME ).placeholder}
		/>
		<TouchableInput
			testID="arrival-time-input"
			containerStyle={styles.withBottomSpace}
			required
			label={presenter.form.$( ARRIVAL_TIME ).label}
			value={presenter.arrivalTime}
			onPress={() => presenter.onArrivalTimePressed()}
			placeholder={presenter.form.$( DEPARTURE_TIME ).placeholder}
		/>
	</HorizontalPadding>
);

FlightForm.propTypes = {
	testID: PropTypes.string,
	presenter: PropTypes.instanceOf( FlightFormPresenter ).isRequired
};

FlightForm.defaultProps = {
	testID: 'flightForm-component'
};

export default observer( FlightForm );
