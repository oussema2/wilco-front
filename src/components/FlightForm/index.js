export { default as FlightForm } from './FlightForm';
