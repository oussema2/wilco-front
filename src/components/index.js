export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
