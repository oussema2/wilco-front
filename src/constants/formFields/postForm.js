export const TITLE = 'title';
export const MESSAGE = 'message';
export const VISIBILITY = 'visibility';
export const FLIGHT = 'flight';
export const FROM = 'from';
export const TO = 'to';
export const DEPARTURE_TIME = 'departureTime';
export const ARRIVAL_TIME = 'arrivalTime';
