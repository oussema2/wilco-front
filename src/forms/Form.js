import MobxReactForm from 'mobx-react-form';
import dvr from '../plugins/dvr';

export default class Form extends MobxReactForm {
	constructor(
		{ fields = [] } = {},
		{
			hooks,
			options = {
				validateOnChange: true,
				showErrorsOnChange: false,
				validationDebounceWait: 0
			},
			plugins = dvr
		} = {}
	) {
		super( { fields }, { options, hooks, plugins } );
	}
}
