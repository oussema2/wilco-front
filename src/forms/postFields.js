import {
	TITLE, MESSAGE, VISIBILITY, FLIGHT, FROM, TO, DEPARTURE_TIME, ARRIVAL_TIME
} from '../constants/formFields/postForm';

const commonFields = [ {
	name: TITLE,
	label: 'Title',
	placeholder: 'Add a title'
}, {
	name: MESSAGE,
	label: 'Message / Trip notes',
	placeholder: 'Share an activity or experience\n'
			+ 'Request feedback\n'
			+ 'Post a PIREP'
}, {
	name: VISIBILITY,
	label: 'Post Visibility',
	placeholder: 'Select the visibility'
} ];

const fields = [
	...commonFields,
	{
		name: FLIGHT,
		label: 'Flight',
		fields: [ {
			name: FROM,
			label: 'From',
			placeholder: 'Origin code',
			rules: 'required|between:3,4'
		}, {
			name: TO,
			label: 'To',
			placeholder: 'Destination code',
			rules: 'required|between:3,4'
		}, {
			name: DEPARTURE_TIME,
			label: 'Departure',
			placeholder: 'Pick the date and time',
			rules: 'required'
		}, {
			name: ARRIVAL_TIME,
			label: 'Arrival',
			placeholder: 'Pick the date and time',
			rules: 'required'
		} ]
	}
];

export default fields;
export { commonFields as editFields };
