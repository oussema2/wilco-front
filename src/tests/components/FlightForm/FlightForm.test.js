import * as React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { FlightForm } from '../../../components/FlightForm';
import Form from '../../../forms/Form';
import { FLIGHT } from '../../../constants/formFields/postForm';
import postFields from '../../../forms/postFields';

describe( 'FlightForm', () => {
	const testID = 'test-id';
	const form = new Form( { fields: postFields, hooks: {} } ).$( FLIGHT );
	const presenter = {
		departureTime: 'Yesterday, 21:45',
		arrivalTime: 'Today, 07:38',
		onDepartureTimePressed: jest.fn(),
		onArrivalTimePressed: jest.fn(),
		form
	};
	let component;

	beforeEach( () => {
		component = render( <FlightForm testID={testID} presenter={presenter} /> );
	} );

	describe( 'with the default props', () => {
		it( 'renders the FlightForm component correctly', () => {
			expect( component.queryByTestId( testID ) ).not.toBeNull();
			expect( component.queryByTestId( 'post-manually-form-title' ) ).toHaveTextContent( 'Flight data - Add flight manually' );
			expect( component.queryByTestId( 'from-input' ) ).toBeDefined();
			expect( component.queryByTestId( 'to-input' ) ).toBeDefined();
			expect( component.queryByTestId( 'departure-time-input' ) ).toHaveTextContent( 'Yesterday, 21:45' );
			expect( component.queryByTestId( 'arrival-time-input' ) ).toHaveTextContent( 'Today, 07:38' );
			expect( component ).toMatchSnapshot();
		} );
	} );

	// TODO: fix this test so that it adapts to the validation on blur.

	// eslint-disable-next-line jest/no-commented-out-tests
	// describe( 'when the form has validation errors', () => {
	// 	eslint-disable-next-line jest/no-commented-out-tests
	// 	it( 'shows the messages of the errors', async () => {
	// 		const fromInput = component.queryAllByTestId( 'input' )[ 0 ];
	// 		const toInput = component.queryAllByTestId( 'input' )[ 1 ];
	// 		await act( async () => {
	// 			fireEvent( fromInput, 'change', 'invalid origin' );
	// 			fireEvent( toInput, 'change', 'ABC' );
	// 			fireEvent( toInput, 'change', '' );
	// 		} );
	//		eslint-disable-next-line max-len
	// 		expect( component.queryByText( 'This field must be between 3 and 4 characters.' ) ).not.toBeNull();
	// 		expect( component.queryByText( 'This field is mandatory.' ) ).not.toBeNull();
	// 	} );
	// } );

	describe( 'when the departure time field is pressed', () => {
		it( 'calls the presenter\'s onDepatureTimePressed callback', () => {
			const departureTimeField = component.getByTestId( 'departure-time-input' );
			fireEvent.press( departureTimeField );
			expect( presenter.onDepartureTimePressed ).toHaveBeenCalledTimes( 1 );
		} );
	} );

	describe( 'when the arrival time field is pressed', () => {
		it( 'calls the presenter\'s onArrivalTimePressed callback', () => {
			const arrivalTimeField = component.getByTestId( 'arrival-time-input' );
			fireEvent.press( arrivalTimeField );
			expect( presenter.onArrivalTimePressed ).toHaveBeenCalledTimes( 1 );
		} );
	} );
} );
