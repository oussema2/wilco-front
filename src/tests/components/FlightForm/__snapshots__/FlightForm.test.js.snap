// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightForm with the default props renders the FlightForm component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 20,
      },
      Object {
        "marginTop": 25,
      },
    ]
  }
  testID="test-id"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "NotoSansJP-Black",
          "fontSize": 16,
          "fontWeight": undefined,
          "letterSpacing": -0.3,
          "lineHeight": 22,
          "minHeight": 22,
          "textTransform": "none",
        }
      }
      testID="post-manually-form-title"
    >
      Flight data - Add flight manually
    </Text>
    <Text
      style={
        Object {
          "color": "#D60A50",
          "fontSize": 16,
        }
      }
      testID="required-testID"
    >
       *
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 25,
      }
    }
  />
  <View
    style={
      Object {
        "marginBottom": 20,
        "minHeight": 98,
      }
    }
    testID="from-input"
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242424",
            "fontFamily": "NotoSansJP-Black",
            "fontSize": 16,
            "fontWeight": undefined,
            "letterSpacing": -0.3,
            "lineHeight": 22,
            "marginBottom": 4,
          }
        }
        testID="input-label"
      >
        From
      </Text>
      <Text
        style={
          Object {
            "color": "#D60A50",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "marginLeft": 4,
          }
        }
        testID="input-required"
      >
        *
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F2F2F2",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 5,
              "minHeight": 52,
              "paddingHorizontal": 12,
              "textAlignVertical": "center",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            autoCapitalize="characters"
            autoCorrect={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Origin code"
            placeholderTextColor="#A0A0A0"
            returnKeyType="default"
            scrollEnabled={true}
            secureTextEntry={false}
            selectionColor="#12A3BE"
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 14,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "minHeight": 40,
                "paddingVertical": 0,
              }
            }
            testID="input"
            textAlignVertical="center"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#D60A50",
              "fontFamily": "NotoSansJP-Regular",
              "fontSize": 12,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 16,
              "margin": 0,
              "marginLeft": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
        "minHeight": 98,
      }
    }
    testID="to-input"
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#242424",
            "fontFamily": "NotoSansJP-Black",
            "fontSize": 16,
            "fontWeight": undefined,
            "letterSpacing": -0.3,
            "lineHeight": 22,
            "marginBottom": 4,
          }
        }
        testID="input-label"
      >
        To
      </Text>
      <Text
        style={
          Object {
            "color": "#D60A50",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "marginLeft": 4,
          }
        }
        testID="input-required"
      >
        *
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F2F2F2",
              "borderBottomWidth": 0,
              "borderColor": "#86939e",
              "borderRadius": 8,
              "flexDirection": "row",
              "marginBottom": 5,
              "minHeight": 52,
              "paddingHorizontal": 12,
              "textAlignVertical": "center",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            autoCapitalize="characters"
            autoCorrect={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Destination code"
            placeholderTextColor="#A0A0A0"
            returnKeyType="default"
            scrollEnabled={true}
            secureTextEntry={false}
            selectionColor="#12A3BE"
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 14,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "minHeight": 40,
                "paddingVertical": 0,
              }
            }
            testID="input"
            textAlignVertical="center"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#D60A50",
              "fontFamily": "NotoSansJP-Regular",
              "fontSize": 12,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 16,
              "margin": 0,
              "marginLeft": 0,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 98,
        "marginBottom": 20,
      }
    }
    testID="departure-time-input"
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "NotoSansJP-Black",
            "fontSize": 16,
            "fontWeight": undefined,
            "letterSpacing": -0.3,
            "lineHeight": 22,
            "marginBottom": 4,
          }
        }
        testID="touchable-input-label"
      >
        Departure
      </Text>
      <Text
        style={
          Object {
            "color": "#D60A50",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "marginLeft": 4,
          }
        }
        testID="touchable-input-required"
      >
        *
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 52,
            "paddingHorizontal": 12,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000000",
              "flex": 1,
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 14,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
            }
          }
          testID="touchable-input-text"
        >
          Yesterday, 21:45
        </Text>
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 23,
                "width": 14,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../src/assets/icons/chevronRight.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="touchable-input-icon"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#D60A50",
            "fontFamily": "NotoSansJP-Regular",
            "fontSize": 12,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 16,
          }
        }
        testID="touchable-input-error"
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 98,
        "marginBottom": 20,
      }
    }
    testID="arrival-time-input"
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "NotoSansJP-Black",
            "fontSize": 16,
            "fontWeight": undefined,
            "letterSpacing": -0.3,
            "lineHeight": 22,
            "marginBottom": 4,
          }
        }
        testID="touchable-input-label"
      >
        Arrival
      </Text>
      <Text
        style={
          Object {
            "color": "#D60A50",
            "fontFamily": "NotoSansJP-Bold",
            "fontSize": 14,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 20,
            "marginLeft": 4,
          }
        }
        testID="touchable-input-required"
      >
        *
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F2F2",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 52,
            "paddingHorizontal": 12,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000000",
              "flex": 1,
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 14,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
            }
          }
          testID="touchable-input-text"
        >
          Today, 07:38
        </Text>
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "height": 23,
                "width": 14,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../src/assets/icons/chevronRight.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="touchable-input-icon"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#D60A50",
            "fontFamily": "NotoSansJP-Regular",
            "fontSize": 12,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 16,
          }
        }
        testID="touchable-input-error"
      >
        
      </Text>
    </View>
  </View>
</View>
`;
